### ISSUE TITLE
FEATURE: Improve alpha blending for X-ray

### ISSUE THREAD

**Issue by xeolabs** [xeokit/xeokit-sdk#1251]
We currently use the same blending modes for Selection, X-Ray and highlighting effects. These modes seem not to make X-ray  transparent enough, where it's always too dark.

Solution could be to 

* allow different blending modes for for X-Ray, Selection and Highlight, 
* expose ability to select mode for each of these effects within the API, or 
* always apply the nicest blending mode for X-ray, leaving blending mode for Selection and Highlight as they are currently

## Currently

X-Ray is too dark with the current blending modes.

![Screenshot from 2023-11-20 19-55-00](https://github.com/xeokit/xeokit-sdk/assets/83100/5b0476c5-b057-4798-af67-54d95ab42498)

````javascript
gl.blendEquation(gl.FUNC_ADD);
gl.blendFuncSeparate(gl.SRC_ALPHA, gl.ONE_MINUS_SRC_ALPHA, gl.ONE, gl.ONE_MINUS_SRC_ALPHA);
````

## Experimental improvement

X-ray looks better with the blending mode shown below.

![Screenshot from 2023-11-20 19-50-06](https://github.com/xeokit/xeokit-sdk/assets/83100/bba3064c-082e-4442-916c-b7ec48969d23)

````javascript
gl.blendEquation(gl.FUNC_ADD);
gl.blendFunc(gl.SRC_ALPHA, gl.ONE_MINUS_SRC_ALPHA);
````

**Comment by pbbr** [comment]
@xeolabs This is a very useful adjustment for X-Ray mode. But we have recognized, that the following mode doesn't work correctly when the [background / canvas is transparent](https://xeokit.github.io/xeokit-sdk/docs/class/src/viewer/scene/canvas/Canvas.js~Canvas.html#instance-member-transparent). In this case, transparent objects getting colored from the canvas background color.
```
gl.blendEquation(gl.FUNC_ADD);
gl.blendFunc(gl.SRC_ALPHA, gl.ONE_MINUS_SRC_ALPHA);
```
Maybe there's a better blending mode / function to achieve the desired result?

**Comment by xeolabs** [comment]
@pbbr I've done lots of experimentation with blending modes but haven't yet found one that works perfectly for all cases. If anyone has suggestions they would be most welcome.

**Comment by pbbr** [comment]
@xeolabs Unfortunately, we didn't find a better configuration too. But what we see is, that with this blend function, transparent objects look very "bright" in Safari (Edge and Chrome are correct, both on Windows and Mac). 

Safari:
<img width="1440" alt="Bild" src="https://github.com/xeokit/xeokit-sdk/assets/79220314/c92b4688-aa03-44b8-b6a7-1618da1d5803">

Edge / Chrome:
![image](https://github.com/xeokit/xeokit-sdk/assets/79220314/1185562f-2e0e-437e-97e6-219e2a40d926)


Interesting is, that a captured snapshot looks correct. Do you have an idea how to change this?

**Comment by xeolabs** [comment]
@pbbr I haven't looked at this issue lately on Safari, so nothing to report yet. Do you represent one of our enterprise users? If so, then if we could track this via our Jira then we can ensure that it gets prioritized properly

**Comment by pbbr** [comment]
@xeolabs Yes, we are. But we don't have access to Jira. How we can get this?

**Comment by xeolabs** [comment]
Could you please contact Wlod via this link, and he should be able to set you up with Jira access: [info@creoox.com](mailto:info@creoox.com)
